### PR TITLE
Porting ackermann steering controller to ros2

### PR DESCRIPTION
**Pull request by EXPX3**

### CODE REVIEW

**Comment by destogl** [comment]
@ijnek can you take a look at this PR? Would this Ackerman controller be also applicable for you? Do we have to add some functionality here?

**Comment by willcbaker** [comment]
+1 friendly bump, is there any outstanding work to be done to land these controllers?

**Comment by ijnek** [comment]
I was going to take a look at this but I'm not quite finding the time for it... would someone else be able to review this?

**Comment by bmagyar** [comment]
@willcbaker let me help you with that

File: ackermann_steering_controller/test/test_ackermann_steering_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+                   rclcpp::Duration::from_seconds(0.1));
+  ASSERT_TRUE(controller_->wait_for_commands(executor));
+  ASSERT_EQ(old_timestamp,
+            (*(controller_->input_ref_.readFromNonRT()))->header.stamp);
+  EXPECT_TRUE(std::isnan((*reference)->twist.linear.x));
+  EXPECT_TRUE(std::isnan((*reference)->twist.angular.z));
+  // EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.linear.x,
+  // 0.45);
+  // EXPECT_EQ((*(controller_->input_ref_.readFromNonRT()))->twist.angular.z,
+  // 0.0);
```
Comment (bmagyar) [inline ackermann_steering_controller/test/test_ackermann_steering_controller.cpp:245]: commented

**Review by bmagyar — CHANGES_REQUESTED** [review]
73bq3x

**Pull request closed without merging**